### PR TITLE
Bump GitHub Action image to ubuntu 22.04

### PR DESCRIPTION
**Pull request by zhsj**
This needs snapcore/action-build@v1.1.2 to fix network issue. See https://github.com/snapcore/action-build/issues/42

### CODE REVIEW

File: .github/workflows/smoke-test.yml [hunk 1]
```diff
@@ -11,13 +11,27 @@ defaults:
...
   smoke-test:
     name: Snapcraft
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-22.04
     steps:
       - uses: actions/checkout@v2
 
+      - name: Build
+        # https://github.com/snapcore/action-build/releases
+        uses: snapcore/action-build@v1
```
Comment (xnox) [inline .github/workflows/smoke-test.yml:20]: Why is this action downgraded from previous v1.1.1 to v1? don't you want to upgrade to v1.1.2 here?

Comment (zhsj) [reply .github/workflows/smoke-test.yml]: We don't need to pin it to a micro release. https://github.com/snapcore/action-build#readme suggests using `v1` as well.

Comment (xnox) [reply .github/workflows/smoke-test.yml]: ah, ok.

However the build still fails. I wonder if we are missing something to allow unified cgroups to work, or if nested unified cgroups prevent running docker nested.

Comment (zhsj) [reply .github/workflows/smoke-test.yml]: Yeah, I'm moving it to draft. I'll work on it this week.

**Pull request merged**